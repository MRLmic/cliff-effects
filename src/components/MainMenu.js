import React from 'react';
import {
  // Button,
  Container,
  Menu,
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { BetaWarning } from './BetaWarning';


const MainMenu = function ({ snippets }) {
  console.log(snippets)
  return (
    <Container>
      <Menu
        inverted
        secondary
        size='large'>
        
        <Menu.Item><Link to="/">{ snippets.i_homeNav }</Link></Menu.Item>
        <Menu.Item><Link to="/about">{ snippets.i_aboutNav }</Link></Menu.Item>
        <Menu.Item>{ snippets.i_githubNav }</Menu.Item>
        <Menu.Item position='right'>
          {/*<Link to="/login"><Button inverted>Log in</Button></Link>*/}
          {/*<Button as='a' inverted style={{ marginLeft: '0.5em' }}>Sign Up</Button>*/}
        </Menu.Item>
        <BetaWarning />
      </Menu>
    </Container>
  );
};  // End MainMenu(<>)


export { MainMenu };
