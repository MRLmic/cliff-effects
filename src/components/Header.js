import React from 'react';

import { Segment, Container } from 'semantic-ui-react';

import { MainMenu } from './MainMenu';
import { BetaWarning } from './BetaWarning';

class Header extends React.Component {
  render() {
    return (
      <Segment
        id={ `appHeader` }
        className="header-segment"
        inverted
        textAlign='center'
        vertical
        color='teal'>
        <Container>
          <MainMenu snippets={ this.props.snippets } />
          <BetaWarning snippets={ this.props.snippets } />
        </Container>
      </Segment>
    );
  }
}

export default Header;
