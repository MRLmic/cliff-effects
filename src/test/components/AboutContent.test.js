import React from 'react';
import { mount } from 'enzyme';

import { AboutContent } from '../../components/AboutContent';

const snippets = new Proxy({}, {
  get() {
    return 'Some translated text.';
  },
});

describe('<AboutContent>', () => {
  it('matches snapshot', () => {
    expect(mount(<AboutContent snippets={snippets} />)).toMatchSnapshot();
  });
});
