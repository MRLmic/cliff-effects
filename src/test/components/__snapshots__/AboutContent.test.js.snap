// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutContent> matches snapshot 1`] = `
<AboutContent
  snippets={null}
>
  <div>
    <Header
      as="h1"
      className="ac-header"
    >
      <h1
        className="ui header ac-header"
      >
        Some translated text.
      </h1>
    </Header>
    <Header
      as="h3"
    >
      <h3
        className="ui header"
      >
        Some translated text.
      </h3>
    </Header>
    <Message>
      <div
        className="ui message"
      >
        Some translated text.
      </div>
    </Message>
    <p>
      Some translated text.
    </p>
    <Header
      as="h3"
    >
      <h3
        className="ui header"
      >
        Some translated text.
      </h3>
    </Header>
    <p>
      Some translated text.
    </p>
    <p>
      Some translated text.
    </p>
    <ul>
      <li>
        <ExternalLink
          href="https://www.youtube.com/watch?v=BveX_rID4_E"
        >
          <a
            href="https://www.youtube.com/watch?v=BveX_rID4_E"
            target="_blank"
          >
            Some translated text.
          </a>
        </ExternalLink>
      </li>
      <li>
        <ExternalLink
          href="http://www.nccp.org/projects/files/NCCP_CO_presentation07.pdf"
        >
          <a
            href="http://www.nccp.org/projects/files/NCCP_CO_presentation07.pdf"
            target="_blank"
          >
            Some translated text.
          </a>
        </ExternalLink>
      </li>
      <li>
        <ExternalLink
          href="https://www.umb.edu/editor_uploads/images/centers_institutes/center_social_policy/The_Road_to_the_Cliff_Edge_08.16.17.pdf"
        >
          <a
            href="https://www.umb.edu/editor_uploads/images/centers_institutes/center_social_policy/The_Road_to_the_Cliff_Edge_08.16.17.pdf"
            target="_blank"
          >
            Some translated text.
          </a>
        </ExternalLink>
      </li>
    </ul>
    <Header
      as="h3"
    >
      <h3
        className="ui header"
      >
        Some translated text.
      </h3>
    </Header>
    <p>
      Some translated text.
    </p>
    <Message>
      <div
        className="ui message"
      >
        Some translated text.
      </div>
    </Message>
    <Header
      as="h3"
    >
      <h3
        className="ui header"
      >
        Some translated text.
      </h3>
    </Header>
    <p>
      Some translated text.
    </p>
    <p>
      Some translated text.
    </p>
    <p>
      Some translated text.
    </p>
  </div>
</AboutContent>
`;
